apis/projectcontour/v1/httpproxy.go

```diff
@@ -244,6 +244,64 @@ type VirtualHost struct {
 	// The policy for rate limiting on the virtual host.
 	// +optional
 	RateLimitPolicy *RateLimitPolicy `json:"rateLimitPolicy,omitempty"`
+	// Providers to use for verifying JSON Web Tokens (JWTs) on the virtual host.
+	// +optional
+	JWTProviders []JWTProvider `json:"jwtProviders,omitempty"`
+}
+
+// JWTProvider defines how to verify JWTs on requests.
+type JWTProvider struct {
+	// Unique name for the provider.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	Name string `json:"name"`
+
+	// Whether the provider should apply to all
+	// routes in the HTTPProxy/its includes by
+	// default. At most one provider can be marked
+	// as the default. If no provider is marked
+	// as the default, individual routes must explicitly
+	// identify the provider they require.
+	// +optional
+	Default bool `json:"default,omitempty"`
+
+	// Issuer that JWTs are required to have in the "iss" field.
+	// If not provided, JWT issuers are not checked.
+	// +optional
+	Issuer string `json:"issuer,omitempty"`
+
+	// Audiences that JWTs are allowed to have in the "aud" field.
+	// If not provided, JWT audiences are not checked.
+	// +optional
+	Audiences []string `json:"audiences,omitempty"`
+
+	// Remote JWKS to use for verifying JWT signatures.
+	// +kubebuilder:validation:Required
+	RemoteJWKS RemoteJWKS `json:"remoteJWKS"`
+}
+
+// RemoteJWKS defines how to fetch a JWKS from an HTTP endpoint.
+type RemoteJWKS struct {
```

Comment:
TLS validation fields & impl still need to be added, I was thinking I'd do that in a follow-up PR.